engine/table/src/main/java/io/deephaven/engine/table/impl/select/FilterSelectColumn.java

```diff
@@ -0,0 +1,271 @@
+                    "Cannot use a filter with column Vectors (_ syntax) in select, view, update, or updateView: "
+                            + filter);
+        }
+        if (filter.hasVirtualRowVariables()) {
+            throw new UncheckedTableException(
+                    "Cannot use a filter with virtual row variables (i, ii, or k) in select, view, update, or updateView: "
+                            + filter);
+        }
+
+        return filter.getColumns();
+    }
+
+    @Override
+    public Class<?> getReturnedType() {
+        return Boolean.class;
+    }
+
+    @Override
+    public Class<?> getReturnedComponentType() {
+        return null;
+    }
+
+    @Override
+    public List<String> getColumns() {
+        return filter.getColumns();
+    }
+
+    @Override
+    public List<String> getColumnArrays() {
+        /* This should always be empty, because initDef throws when arrays or ii and friends are used. */
+        return List.of();
+    }
+
+    @Override
+    public boolean hasVirtualRowVariables() {
+        /* This should always be false, because initDef throws when arrays or ii and friends are used. */
+        return false;
+    }
+
+    @NotNull
+    @Override
+    public ColumnSource<Boolean> getDataView() {
+        return new ViewColumnSource<>(Boolean.class, new FilterFormula(), false);
+    }
+
+    @NotNull
+    @Override
+    public ColumnSource<?> getLazyView() {
+        return getDataView();
+    }
+
+    @Override
+    public String getName() {
+        return destName;
+    }
+
+    @Override
+    public MatchPair getMatchPair() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public final WritableColumnSource<?> newDestInstance(final long size) {
+        return SparseArrayColumnSource.getSparseMemoryColumnSource(size, Boolean.class);
+    }
+
+    @Override
+    public final WritableColumnSource<?> newFlatDestInstance(final long size) {
+        return InMemoryColumnSource.getImmutableMemoryColumnSource(size, Boolean.class, null);
+    }
+
+    @Override
+    public boolean isRetain() {
+        return false;
+    }
+
+    @Override
+    public boolean isStateless() {
+        return filter.permitParallelization();
+    }
+
+    @Override
+    public FilterSelectColumn copy() {
+        return new FilterSelectColumn(destName, filter.copy());
+    }
+
+    @Override
+    public void validateSafeForRefresh(BaseTable<?> sourceTable) {
+        filter.validateSafeForRefresh(sourceTable);
+    }
+
+    private class FilterFormula extends Formula {
+        public FilterFormula() {
+            super(null);
+        }
+
+        @Override
+        public Boolean getBoolean(long rowKey) {
+            WritableRowSet filteredIndex = filter.filter(RowSetFactory.fromKeys(rowKey), rowSet, tableToFilter, false);
+            return filteredIndex.isNonempty();
+        }
+
+        @Override
+        public Boolean getPrevBoolean(long rowKey) {
```

Comment:
I think there's no test coverage for `getBoolean` or `getPrevBoolean`.

Reply:
You are right.